pkg/prometheus/statefulset.go

```diff
@@ -614,13 +614,46 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 			thanosArgs = append(thanosArgs, fmt.Sprintf("--log.level=%s", p.Spec.LogLevel))
 		}
 
+		thanosVolumeMounts := []v1.VolumeMount{
+			{
+				Name:      volName,
+				MountPath: storageDir,
+				SubPath:   subPathForStorage(p.Spec.Storage),
+			},
+		}
+
+		envVars := []v1.EnvVar{}
 		if p.Spec.Thanos.GCS != nil {
 			if p.Spec.Thanos.GCS.Bucket != nil {
 				thanosArgs = append(thanosArgs, fmt.Sprintf("--gcs.bucket=%s", *p.Spec.Thanos.GCS.Bucket))
 			}
+			if p.Spec.Thanos.GCS.SecretKey != nil {
+				secretFileName := "service-account.json"
+				if p.Spec.Thanos.GCS.SecretKey.Name != "" {
+					secretFileName = p.Spec.Thanos.GCS.SecretKey.Name
+				}
+				secretDir := "/var/run/secrets/prometheus.io/" + p.Spec.Thanos.GCS.SecretKey.Key
+				envVars = append(envVars, v1.EnvVar{
+					Name:  "GOOGLE_APPLICATION_CREDENTIALS",
+					Value: secretDir + "/" + secretFileName,
```

Comment:
we should use `filepath.Join` for all of the path concatenations we're doing here